### DISCUSSION TITLE
3.0 Release

### DISCUSSION THREAD

**Discussion by andrewgy8** [django-import-export/django-import-export#1284] [General]
Hey! Ive noticed that we have about 3 PR's that are changing the API. I have personally see no issue with this. However I think it would be best that we group them together and release them at the same time in a consolidated 3.0 release. 

I was thinking we could release whats on master at the moment in a 2.x release. And then merge the other API changes, and release another 3.0. 

What do you all think @manelclos @matthewhegarty ? 

If this sounds good to you all, could we mark the PR's with a "Breaking Change" tag so we dont accidentally merge them?

**Comment by manelclos** [comment]
@andrewgy8 totally agree, my https://github.com/django-import-export/django-import-export/pull/1271#issuecomment-830757801 was pointing on that direction.

👍 on tagging them in some way

**Comment by matthewhegarty** [comment]
I wonder if there are any ways around breaking the API - it should be something we only do if we absolutely have to.
But I agree they should be in a 3.0 release.

- #1277 - Am I right in thinking that this doesn't affect the API?  (I tried to ensure it did not)
- #1278 - This does include a breaking change (have tagged it)

**Reply by manelclos** [reply]
> * #1277 - Am I right in thinking that this doesn't affect the API?  (I tried to ensure it did not)

It does not break the API. It introduced a new function `get_filename` that is not needed and I tried to simplify the code here: https://github.com/django-import-export/django-import-export/pull/1282

I've marked my PR as "breaking change" as well.

**Comment by matthewhegarty** [comment]
@andrewgy8 @manelclos Do you think a 3.0 release would be a good opportunity to pick up some outstanding issues?

For example:

- Clear historic issues / fix bugs
- Improve documentation
- New features?
- Improve coverage
- Drop support for older versions
- Anything else?

We've got a bit of momentum going recently, and also we've had a recent offer of help from @thauk-copperleaf 
We could use this thread to get a loose list of potential changes.  I was going to start looking into improving coverage - it's not far off 100% so it should be possible to get close to that with a bit of effort.

Also, it would be good to document the best way for users to get help.  We seem to get a few vague and repeated questions in the issues list from time-to-time.  I generally try to pick up as many [SO](https://stackoverflow.com/questions/tagged/django-import-export) questions as I can.

Any thoughts?

**Comment by matthewhegarty** [comment]
Release 3.0.0 (beta) [available](https://pypi.org/project/django-import-export/3.0.0b0/).  Users are encouraged to download and test and let us know any issues or feedback.

**Comment by matthewhegarty** [comment]
Have been super busy over the summer so have not had time to look again at the 3.0 release.  The beta version has been out for a few months now (also linked from readme, so users know to use it if they want).  I'll look to release 3.0 this week.  Obviously 2.9.0 is still out there so there's no immediate risk to breaking existing installations.

Just wanted to give you a heads up.

**Reply by manelclos** [reply]
Hi @matthewhegarty, thanks for taking the time to push it forward. I've been using the beta version in a couple of projects with no problems and taking benefit of the new fixes and features.

**Comment by matthewhegarty** [comment]
Release 3.0 is out... Thanks to all those who have contributed...

**Reply by andrewgy8** [reply]
Thank you @matthewhegarty for pushing this through! Great work!

**Reply by matthewhegarty** [reply]
Thanks Andrew - I read that the package is in the [top 100](https://blog.devgenius.io/2021-top-100-django-packages-list-during-the-year-92fef0ba79c9) of downloaded Django packages, and in the [top 2k](https://hugovk.github.io/top-pypi-packages/) of python packages, so all of us who have worked on this are making an important contribution to the Django community.  On to release 4 👍

**Reply by andrewgy8** [reply]
Yeah, I read that too. Amazing! 

Hopefully Ill be around more now, as Ive forced Django down the throat of my current org so Im in the thick of it.

Actually, we just installed DjIE into our project, and it worked flawlessly.

**Reply by matthewhegarty** [reply]
Fantastic - it would be great to have you with us again
